src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -87,6 +87,54 @@ public static string GetFullPath(string path)
             return fullPath;
         }
 
+        public static string GetFullPath(string path, string basePath)
+        {
+            if (path == null)
+                throw new ArgumentException(nameof(path));
+
+            if (basePath == null)
+                throw new ArgumentNullException(nameof(basePath));
+
+            if (!IsPathFullyQualified(basePath))
+                throw new ArgumentException(SR.Arg_BasePathNotFullyQualified);
+
+            if (basePath.Contains('\0') || path.Contains('\0'))
+                throw new ArgumentException(SR.Argument_InvalidPathChars);
+
+            if (IsPathFullyQualified(path))
+                return GetFullPath(path);
+
+            int length = path.Length;
+
+            if (PathInternal.IsDevice(basePath))
+            {
+                // paths starting with \\?\ & \\.\
+                return RemoveRelativeSegments(CombineNoChecks(basePath, path), PathInternal.GetRootLength(basePath));
+            }
+            else if ((length >= 1 && PathInternal.IsDirectorySeparator(path[0])))
+            {
+                // path is current drive rooted i.e. starts with \
+                return RemoveRelativeSegments(CombineNoChecks(GetPathRoot(basePath), path.Substring(1)));
+            }
+            else if (length >= 2 && PathInternal.IsValidDriveChar(path[0]) && path[1] == PathInternal.VolumeSeparatorChar)
+            {
+                Debug.Assert(length == 2 || !PathInternal.IsDirectorySeparator(path[2]));
+                
+                // same and different specific drive rooted path
+                if (GetPathRoot(path.AsReadOnlySpan()) == GetPathRoot(basePath.AsReadOnlySpan()))
```

Comment:
You'll need to do `SequenceEqual()` to see if they are actually equivalent. `==` tells you if they are the same length and point to the same memory.

Reply:
Getpath root return "C:" and "C:\" for the input "C:" and "C:\" respectively. So I changed this check to ```GetPathRoot(path.AsReadOnlySpan())[0] == GetPathRoot(basePath.AsReadOnlySpan())[0]``` 
Is this acceptable ?.  Its working as required

Reply:
D'oh. Sorry. I mentioned this earlier and didn't fix it in the copy I made. You cannot use == to compare the contents of spans. Use SequenceEqual(). That is the right fix.

Reply:
Sequence Equal is not available in coreclr

Reply:
Get root path is returning different roots for "C:foo" and "C:\\foo" so sequence equal will also not work

Reply:
SequenceEqual is an extension method and is avaialable only in corefx. We will have to move the its implementation.

Reply:
Good catch for the "C:". I also missed the comparison between device and non device paths. The logic is a little complicated, here is rough in:

``` C#
static ReadOnlySpan<char> GetVolumeName(ReadOnlySpan<char> path)
{
    if (!IsFullyQualified(path))
        return default;

    // 3 cases: UNC ("\\server\share"), Device ("\\?\C:\"), or Dos ("C:\")

    ReadOnlySpan<char> root = GetRootPath(path);
    if (!PathInternal.IsDevice(path))
        return TrimTrailingDirectorySeparators(root); // e.g. "C:"


    if (IsUnc(path))
    {
        // Cut from "\\?\UNC\Server\Share" to "Server\Share"
        return TrimTrailingDirectorySeparators(root.Slice(8));
    }
    else
    {
        // Cut from "\\?\C:\" to "C:"
        return TrimTrailingDirectorySeparators(root.Slice(4));
    }
}
```

It is more complicated than it needs to be for this particular usage, but better it be fully correct as it isn't costly and we'll likely need it for other things.

Also, apparently there is a bug in GetPathRoot() that I missed when I wrote it...

```
Path.GetPathRoot(@"\\?\C:\")
"\\\\?\\C:\\"
Path.GetPathRoot(@"\\?\UNC\Server\Share\")
"\\\\?\\UNC\\Server\\Share"
Path.GetPathRoot(@"\\?\UNC\Server\Share\Foo")
"\\\\?\\UNC\\Server\\Share"
Path.GetPathRoot(@"\\?\FooDrive\Bar")
"" <-- Should be \\?\FooDrive
Path.GetPathRoot(@"\\?\FooDrive")
"" <-- Should be \\?\FooDrive
```

If you want to try and fix that, please give it a shot. :) (In another PR)

I didn't spend much time thinking through UNCs for this API, but we should probably do a follow up change to make sure we're supporting UNC base paths reasonably well (both device and . We may already be ok with the current logic- definitely write some tests)

Reply:
Btw: IsUnc is starts with `\\` and not `\\.` or `\\?` OR starts with `\\.\UNC\` or `\\?\UNC\`.

Reply:
are there any existing function for IsUnc and TrimTrailingDirectorySeparators ?

Reply:
@jkotas should move sequenceEqual from corefx to shared folder. As it will be easier to check equality of spans in coreclr ?

Reply:
Yes, it is on @ahsonkhan TODO list. For now, use StringSpanHelpers.Equals. Once it is moved, all calls to it will be replaced with SequenceEqual